common/src/androidTest/java/com/microsoft/identity/common/crypto/AndroidWrappedKeyLoaderTest.java

```diff
@@ -0,0 +1,171 @@
+        Assert.assertEquals(generatedKeyPair.getPublic(), keyPairReadFromKeyStore.getPublic());
+
+        // Delete
+        AndroidKeyStoreUtil.deleteKey(alias);
+
+        // Read again - should be empty.
+        Assert.assertNull(AndroidKeyStoreUtil.readKey(alias));
+    }
+
+    private AlgorithmParameterSpec getMockKeyPairGeneratorSpec(final String alias){
+        final Date startDate = new Date();
+        final Date endDate =  new Date(startDate.getTime() + 1000000);
+
+        return new KeyPairGeneratorSpec.Builder(context)
+                .setAlias(alias)
+                .setSubject(new X500Principal("CN=SOME_CN, OU=SOME_OU"))
+                .setSerialNumber(BigInteger.ONE)
+                .setStartDate(startDate)
+                .setEndDate(endDate)
+                .build();
+    }
+
+    @Test
+    public void testGenerateKey() throws ClientException {
+        final AndroidWrappedKeyLoader loader = new AndroidWrappedKeyLoader(context, null);
+        final SecretKey secretKey = loader.generateRandomKey();
+
+        Assert.assertEquals(AES256KeyLoader.AES_ALGORITHM, secretKey.getAlgorithm());
+    }
+
+    @Test
+    public void testReadKeyDirectly() throws ClientException {
+        final AndroidWrappedKeyLoader loader = initKeyLoaderWithKeyEntry();
+        final SecretKey secretKey = loader.getKey();
+        final SecretKey storedSecretKey = loader.readSecretKeyFromStorage();
+
+        // They're not the same object!
+        Assert.assertNotSame(secretKey, storedSecretKey);
+
+        Assert.assertEquals(AES256KeyLoader.AES_ALGORITHM, secretKey.getAlgorithm());
+        Assert.assertEquals(AES256KeyLoader.AES_ALGORITHM , storedSecretKey.getAlgorithm());
+
+        Assert.assertArrayEquals(secretKey.getEncoded(), storedSecretKey.getEncoded());
+        Assert.assertEquals(secretKey.getFormat(), storedSecretKey.getFormat());
+    }
+
+    @Test
+    public void testLoadKey() throws ClientException {
+        // Nothing exists. This load key function should generate a key if the key hasn't exist.
+        Assert.assertNull(AndroidKeyStoreUtil.readKey(AndroidWrappedKeyLoader.KEY_STORE_CERT_ALIAS));
+        Assert.assertNull(FileUtil.readFromFile(context,
+                AndroidWrappedKeyLoader.KEY_FILE_PATH,
+                AndroidWrappedKeyLoader.KEY_FILE_SIZE));
+
+        final AndroidWrappedKeyLoader loader = new AndroidWrappedKeyLoader(context, null);
+        final SecretKey secretKey = loader.getKey();
+        Assert.assertNotNull(loader.mCachedKey);
+
+        Assert.assertEquals(AES256KeyLoader.AES_ALGORITHM, secretKey.getAlgorithm());
+        Assert.assertArrayEquals(secretKey.getEncoded(), loader.mCachedKey.getEncoded());
+        Assert.assertEquals(secretKey.getFormat(), loader.mCachedKey.getFormat());
+    }
+
+    @Test
+    public void testLoadDeletedKeyStoreKey() throws ClientException {
+        final AndroidWrappedKeyLoader loader = initKeyLoaderWithKeyEntry();
```

Comment:
Don't really like these variables to be called `loader` here and everywhere...would prefer something like `keyLoader`